### PR TITLE
Add Eagle RWB Blast Support

### PR DESCRIPTION
**Pull request by Ignose**

### CODE REVIEW

File: scripts/bountiful.ash [hunk 1]
```diff
@@ -665,10 +671,15 @@ boolean hunt_bounty(bounty b) {
             mall_price(CONTENT_ITEMS[b.location]) <= maxSpecial)) {
     if(useBan)
       buy_banishers();
-
+    
+    //If have eagle and don't have RWB, use eagle; get those free copies.
+    if(have_familiar($familiar[Patriotic Eagle]) && !have_effect($effect[Everything Looks Red, White and Blue]))
```
Comment (Alium58) [inline scripts/bountiful.ash]: also check it isn't already our active fam, similar to nosy nose below

File: scripts/bountiful.ash [hunk 2]
```diff
@@ -665,10 +671,15 @@ boolean hunt_bounty(bounty b) {
...
+      use_familiar($familiar[Patriotic Eagle]);
+    else {
```
Comment (s-k-z) [inline scripts/bountiful.ash]:
```suggestion
    if(have_familiar($familiar[Patriotic Eagle]) && have_effect($effect[Everything Looks Red, White and Blue]) < 1) {
      use_familiar($familiar[Patriotic Eagle]);
    } else {
```
Using `!have_effect()` results in `"!" operator requires a boolean value`

**Review by s-k-z — CHANGES_REQUESTED** [review]

File: scripts/bountiful.ash (hunk 2 above)
Comment (Ignose) [reply scripts/bountiful.ash]: Fixed.

File: scripts/bountiful.ash [hunk 3]
```diff
@@ -335,6 +335,12 @@ string addBountyToQueue(monster opp, boolean speculate) {
     return "skill Motif";
   }
 
+  //Why not Red White and Blue?
+  if(!have_effect($effect[Everything Looks Red, White and Blue]) && have_skill($skill[%fn, fire a Red White and Blue Blast]))
```
Comment (Alium58) [inline scripts/bountiful.ash]: `!have_effect($effect[Everything Looks Red, White and Blue])`
should be
`have_effect($effect[Everything Looks Red, White and Blue]) == 0`

not intuitive, but `have_effect` returns an int, not a boolean

File: scripts/bountiful.ash (hunk 2 above)
Comment (Alium58) [inline scripts/bountiful.ash]: make this an `else if` by moving line 680 up to here

File: scripts/bountiful.ash [hunk 4]
```diff
@@ -335,6 +335,12 @@ string addBountyToQueue(monster opp, boolean speculate) {
     return "skill Motif";
   }
 
+  //Why not Red White and Blue?
```
Comment (Alium58) [inline scripts/bountiful.ash]: say its from the eagle iotm

File: scripts/bountiful.ash (hunk 3 above)
Comment (Alium58) [reply scripts/bountiful.ash]: did this in the cli
`ash $skill[%fn, fire a Red White and Blue Blast]`

It suggested:
`%fn\, fire a Red\, White and Blue Blast`

Updating to that would remove some nuance messages from mafia

**Review by Alium58 — CHANGES_REQUESTED** [review]
thank you! Have a few comments but all are minor

File: scripts/bountiful.ash (hunk 3 above)
Comment (Ignose) [reply scripts/bountiful.ash]: Should I change the 'return' as well?

Comment (Alium58) [reply scripts/bountiful.ash]: No that one should be fine

File: scripts/bountiful.ash (hunk 4 above)
Comment (Ignose) [reply scripts/bountiful.ash]: Updated

File: scripts/bountiful.ash (hunk 2 above)
Comment (Ignose) [reply scripts/bountiful.ash]: Updated

File: scripts/bountiful.ash (hunk 1 above)
Comment (Ignose) [reply scripts/bountiful.ash]: Updated

File: scripts/bountiful.ash (hunk 3 above)
Comment (Ignose) [reply scripts/bountiful.ash]: Excellent. Updated.

File: scripts/bountiful.ash [hunk 5]
```diff
@@ -665,10 +671,15 @@ boolean hunt_bounty(bounty b) {
             mall_price(CONTENT_ITEMS[b.location]) <= maxSpecial)) {
     if(useBan)
       buy_banishers();
-
-    // use Nosy Nose to add copies of bounty to the queue
-    if(have_familiar($familiar[Nosy Nose]) && my_familiar() != $familiar[Nosy Nose])
+    
+    //If have eagle and don't have RWB, use eagle; get those free copies.
+    if(have_familiar($familiar[Patriotic Eagle]) && have_effect($effect[Everything Looks Red, White and Blue]) < 1 
+      && my_familiar() != $familiar[Nosy Nose]) {
```
Comment (Alium58) [inline scripts/bountiful.ash]: patriotic eagle instead of nose

Comment (Ignose) [reply scripts/bountiful.ash]: And fixed.

**Review by Alium58 — APPROVED** [review]
Thank you!

**Pull request merged**